dot/peerset/peerset.go

```diff
@@ -665,20 +693,13 @@ func (ps *PeerSet) disconnect(setIdx int, reason DropReason, peers ...peer.ID) e
 }
 
 // start handles all the action for the peerSet.
-func (ps *PeerSet) start(ctx context.Context, actionQueue chan action) {
-	ps.actionQueue = actionQueue
-
-	ps.resultMsgCh = make(chan Message, msgChanSize)
-	go ps.doWork(ctx)
+func (ps *PeerSet) start(ctx context.Context) {
+	go ps.periodicallyAllocateSlots(ctx)
 }
```

Comment:
Maybe remove this and call directly `go ps.periodicallyAllocateSlots(ctx)` from the caller calling `.start(ctx)`?

So it's more explicit it's launching a goroutine, and also saves one function that's not so needed imo.